__tests__/app.test.js

```diff
@@ -312,4 +312,117 @@ describe("GET comments by review ID", () => {
+            username: "philippaclaire9",
+            body: "Where is my head?"
+        };
+
+        test("When a valid comment is added it is returned as an object", () => {
+            return request(app)
+                .post("/api/reviews/1/comments")
+                .send(testComment)
+                .expect(201)
+                .then(({ body }) => {
+                    const newComment = body.comment;
+                    expect(newComment.author).toEqual(testComment.username);
+                });
+        });
+
+        test("Ensure that only the posted comment is returned", () => {
+            return request(app)
+                .post("/api/reviews/2/comments")
+                .send(testComment)
+                .expect(201)
+                .then(({ body }) => {
+                    const newComment = body.comment;
+                    expect(newComment.author).toEqual(testComment.username);
+                });
+        });
+
+        test("ensure that when extra information is given it is ignored", () => {
+            testComment.extra = "Extra unncessary info";
+            return request(app)
+                .post("/api/reviews/1/comments")
+                .send(testComment)
+                .expect(201)
+                .then(({ body }) => {
+                    const newComment = body.comment;
+                    expect(newComment.author).toEqual(testComment.username);
+                });
+        });
+    });
+
+    describe("sad path :(", () => {
+        test("when no information is sent send back the appropriate error", () => {
+            return request(app)
+                .post("/api/reviews/1/comments")
+                .expect(400)
+                .then(({ body }) => {
+                    expect(body.msg).toEqual("Please provide a username.");
+                });
+        });
+
+        test("when no username is sent send back the appropriate error", () => {
+            return request(app)
+                .post("/api/reviews/1/comments")
+                .send({ body: "test text" })
+                .expect(400)
+                .then(({ body }) => {
+                    expect(body.msg).toEqual("Please provide a username.");
+                });
+        });
+
+        test("when no body is sent send back the appropriate error", () => {
+            return request(app)
+                .post("/api/reviews/1/comments")
+                .send({ username: "username goes here" })
+                .expect(400)
+                .then(({ body }) => {
+                    expect(body.msg).toEqual("Please provide a body.");
+                });
+        });
+
+        test("when an invalid username is sent send back the appropriate error", () => {
+            return request(app)
+                .post("/api/reviews/1/comments")
+                .send({
+                    username: "invalid username",
+                    body: "Oh, here's my head!"
+                })
+                .expect(400)
```

Comment:
404 not found is more appropriate here than bad request, its like a comment id being valid but just non-existent